jdaviz/configs/default/plugins/subset_tools/subset_tools.py

```diff
@@ -238,8 +238,9 @@ def get_regions(self, region_type=None, list_of_subset_labels=None,
                                        include_sky_region=reg_type == 'sky_region',
                                        use_display_units=use_display_units)
 
-        labels = list_of_subset_labels or subsets.keys()
-        labels = [labels] if not hasattr(labels, '__iter__') else labels
+        labels = list_of_subset_labels or list(subsets.keys())
+        if not isinstance(labels, (list, np.ndarray, tuple)):
```

Comment:
Same question as https://github.com/spacetelescope/jdaviz/pull/3498/files#r1992116614
```suggestion
        if isinstance(labels, str):
```